### DISCUSSION TITLE
VRR Support

### DISCUSSION THREAD

**Discussion by coatlessali** [YaLTeR/niri#289] [Ideas]
I might just not have looked hard enough (in which case please correct me), but couldn't find a way to enable VRR/adaptive sync. It's an essential feature for good framepacing (especially when playing 60hz content on a 144hz display, or 24hz on 60hz, etc.) and it would be very nice to have available.

**Comment by YaLTeR** [comment]
Yeah, it shouldn't even be too hard to support (need to make sure that the frame clock code doesn't go crazy), but unfortunately, on my setup, what seems to be multiple AMDGPU bugs make it annoying/hard to test, so I haven't implemented it yet.

**Comment by YaLTeR** [comment]
I just added VRR support in 3c6d8062c5c9b31afb5eeacd66a58b19b6566d9f. It seems to kinda work on my setup after some convincing of AMDGPU, but please test if it works for you.

**Reply by YaLTeR** [reply]
Here's the setting: https://github.com/YaLTeR/niri/wiki/Configuration:-Outputs#variable-refresh-rate

**Comment by coatlessali** [comment]
The mentioned `disable-cursor-plane` debug flag was necessary to get smooth mouse movement - otherwise it would be stuck at 48hz. It was like the framerate wouldn't change to match the cursor.

Also on amdgpu. Initial impressions are good otherwise.

**Reply by YaLTeR** [reply]
It seems that cursor plane updates only flip the screen at the lowest refresh rate, yeah. Might be some kernel or driver thing, I'm not very familiar with it.

**Comment by coatlessali** [comment]
Okay, it plays nicely with Xwayland. Played some 24 fps and 60 fps content and it works just fine.

Also, I'd just like to point out that constant modesetting is a symptom of a bad display cable. It's one of the earliest signs that a cable is going bad - or just doesn't provide enough bandwidth.

**Reply by YaLTeR** [reply]
I have constant modesetting on a new monitor connected through HDMI with its own cable. As long as I don't enable VRR, there are no issues. So I am really inclined to think that this is an AMDGPU problem in this case.

**Reply by YaLTeR** [reply]
Just tried a different cable and it did the same thing, so yeah

**Reply by YaLTeR** [reply]
Found a DP -> USB C connector. Through it, VRR works perfectly on this monitor, yay

**Reply by coatlessali** [reply]
Glad it helped. Out of spec cables can definitely be a problem.
